### PR TITLE
Pages & Widgets

### PR DESCRIPTION
**Pull request by PoojaB26**
### Description
Pages & Widgets doc
Linear ticket and [magic word](https://linear.app/docs/github#link-prs) Fixes DEVR-311 & Fixes DEVR-310

## Type of change
- [ ] Typo fix 
- [x] New feature 
- [ ] Removed outdated references

### CODE REVIEW

File: docs/resources/widgets/overview/overview.md
Comment (leighajarett) [inline docs/resources/widgets/overview/overview.md:6]: I think we should specifically start by talking about how in Flutter everything is a widget (pages, atomic elements, layouts, etc) -- but in FlutterFlow thats not really the case, pages arent widgets in FlutterFlow but they are in the generated code. I talk a bit about this in my Flutter connection talk

Comment (leighajarett) [reply docs/resources/widgets/overview/overview.md:6]: Basically I think we need to talk about pages more here 

Also same comment as before, it feels text heavy. I would recommend making a visual in a slide deck (or using some from the Flutter connection talk) to illustrate your point

**Review by leighajarett — CHANGES_REQUESTED** [review]
This is looking really great @PoojaB26!

Left a few comments, but generally my feedback falls into a few categories:

- Unclear what the sequence of this content should be - if concept comes first then we shouldn't use terms we ahven't really introduced yet (i.e. widget)

-Overview doesn't really cover pages, I think we need to tie this in more, or separate pages out of the widgets section

-Lets use code block when referring to widget names (nit)

-Most important: need more visuals. There's a lot of text here. I think we need some visual diagrams to illustrate your points more. You gave lots of good examples, so lets show those examples by creating some diagram in a google slide and taking a screenshot of it or something like that! I would recommend starting with the slides I used for Flutter Connection and tweaking to the examples you're using

File: docs/resources/widgets/Concepts/concepts.md
Comment (PoojaB26) [reply docs/resources/widgets/Concepts/concepts.md]: Yes, makes sense. Will move it around!

File: docs/resources/widgets/overview/overview.md
Comment (PoojaB26) [reply docs/resources/widgets/overview/overview.md]: > but in FlutterFlow thats not really the case, pages arent widgets in FlutterFlow but they are in the generated code

I want to mention this in the generated code page where we show the differences between FlutterFlow's approach and Flutter. I feel it'll be confusing to non Flutter devs to start with "everything in Flutter is a widget but in FlutterFlow its a different approach" because then we are assuming that they are all coming with Flutter understanding. So instead I propose, we just talk about pages as just another entity in FlutterFlow and in generated code page, we talk about the underlying concepts. 

Also there is a pages.md page where we talk about pages, won't that be enough? Also to demo the entire overview of all the entities, maybe that would be covered in Pinkesh's resource hierarchy I think.

File: docs/resources/widgets/Concepts/concepts.md
Comment (PoojaB26) [reply docs/resources/widgets/Concepts/concepts.md]: Added an example section at the end! Let me know how that looks

**Comment by PoojaB26** [comment]
> Unclear what the sequence of this content should be - if concept comes first then we shouldn't use terms we ahven't really introduced yet (i.e. widget)

Agreed, changed! 

> -Overview doesn't really cover pages, I think we need to tie this in more, or separate pages out of the widgets section

Pages is covered separately in pages.md. 
Hierarchy will probably look like this 

Widgets ( = widgets.md)
-- Concepts
-- Pages 
-- Built In Widgets 
-- Components 
-- Custom Widgets 

> -Most important: need more visuals. There's a lot of text here. I think we need some visual diagrams to illustrate your points more. You gave lots of good examples, so lets show those examples by creating some diagram in a google slide and taking a screenshot of it or something like that! I would recommend starting with the slides I used for Flutter Connection and tweaking to the examples you're using

I did add images at places but there were some rendering issues there, but yes, I agree more visuals would be great, so went ahead and added more images, you can go ahead and check them now! 
Also just fyi, instead of Slides, we can create the illustrations in [Figma here](https://www.figma.com/file/X4dJZ2i8C6kpPrcnHmjBkq/Ecommerce-Demo-App?type=design&node-id=534-26&mode=design&t=88uJzDJTWUlIgJ7z-0) so we won't have to take screenshots and can create high quality illustrations. Just letting you know so you can also use the same and all the images can remain in one place for modification later. 🙌🏽 @leighajarett

**Comment by PoojaB26** [comment]
@leighajarett This PR is ready for review, also added some missing sections in Pages How to. 
@pinkeshmars Also have a look and let me know if any missing sections here

**Review by pinkeshmars — COMMENTED** [review]
Seems like there is some issue while reviewing this PR in stackblitz! It just shows blank screen.

Anyways, I think we could add the widget configuration page just like this one (docs/resources/widgets/pages/pages-configuration.md)

**Comment by leighajarett** [comment]
@PoojaB26 I am not able to build the docs site with the current version of the branch. Widgets.md wont render, can you please take a look?

File: docs/resources/widgets/widgets.md
Comment (leighajarett) [inline docs/resources/widgets/widgets.md:4]: I think we need an h1 in here

File: docs/resources/widgets/widgets.md
Comment (leighajarett) [inline docs/resources/widgets/widgets.md:5]: I think we should have a page in the left hand side bar for this page (right now, I have to click on the widget section to get to this page) to be consistent with the projects section. It wasn't really clear to me this existed from the left hand side panel. Perhaps we call it "What's a widget?"

File: docs/resources/widgets/widgets.md
Comment (leighajarett) [inline docs/resources/widgets/widgets.md:44]: I think these should all be bullets. The page design looks a bit weird with them each as their own paragraph

File: docs/resources/widgets/widgets.md
Comment (leighajarett) [inline docs/resources/widgets/widgets.md:53]: "the builder" not your builder

File: docs/resources/widgets/widgets.md
Comment (leighajarett) [inline docs/resources/widgets/widgets.md:74]: We need a consistent way of referring to widgets here. Italics, code, bold -- I don't really care but we should decide one and be consistent.

File: docs/resources/widgets/widgets.md
Comment (leighajarett) [inline docs/resources/widgets/widgets.md:27]: This image seems pretty big, I think we can make it smaller

File: docs/resources/widgets/widgets.md
Comment (leighajarett) [inline docs/resources/widgets/widgets.md:37]: For each of these categories, it would be nice to have a table - with an example of the widget in use / image of the example

Comment (leighajarett) [reply docs/resources/widgets/widgets.md:37]: For example, one row would be for "column widget" - we would have the name / description in one column, and make some visual diagram of a row being used on a page. 

I think we just need some more visuals here, feels too text heavy

File: docs/resources/widgets/widgets.md
Comment (leighajarett) [inline docs/resources/widgets/widgets.md:98]: I'm not sure what you mean here by "main screen widget"

File: docs/resources/widgets/widgets.md
Comment (leighajarett) [inline docs/resources/widgets/widgets.md:100]: We should describe what's actually being shown in the visual below instead of describing some example that we don't see here

File: docs/resources/widgets/widgets.md
Comment (leighajarett) [inline docs/resources/widgets/widgets.md:102]: For all images we need alt text for accessibility purposes

File: docs/resources/widgets/Concepts/key-concepts.md
Comment (leighajarett) [inline docs/resources/widgets/Concepts/key-concepts.md:6]: All pages need an h1

File: docs/resources/widgets/Concepts/key-concepts.md
Comment (leighajarett) [inline docs/resources/widgets/Concepts/key-concepts.md:61]: I'm a bit lost with this visual - can you make it a bit more clear? I think we need to see more of the UI to orient ourselves

File: docs/resources/widgets/Concepts/composing-widgets.md
Comment (leighajarett) [inline docs/resources/widgets/Concepts/composing-widgets.md:6]: I feel like we're repeating a lot of the information from the "widgets.md" file here. I don't think we need to list row, column, etc in both files. I would just keep that information here?

File: docs/resources/widgets/Concepts/composing-widgets.md
Comment (leighajarett) [inline docs/resources/widgets/Concepts/composing-widgets.md:58]: Things like this should be in a bulleted list

File: docs/resources/widgets/Concepts/composing-widgets.md
Comment (leighajarett) [inline docs/resources/widgets/Concepts/composing-widgets.md:3]: I feel like this page looks a bit weird. The images take up a lot of space and theres a lot of blank space. I think it makes sense to put this stuff into a tabular format and make the images much smaller so it's easier to view

Comment (leighajarett) [reply docs/resources/widgets/Concepts/composing-widgets.md:3]: Like I would have a table with row, stack, column each as their own row and the little images you put as a column in the table

File: docs/resources/widgets/pages/pages.md
Comment (leighajarett) [inline docs/resources/widgets/pages/pages.md:2]: Same comment as widgets, I think we need a file nested under the pages section called "Whats a page?" or something. It wasn't clear to me that clicking on Page would bring me to an actual page

File: docs/resources/widgets/pages/pages.md
Comment (leighajarett) [inline docs/resources/widgets/pages/pages.md:17]: We probably want some link back to the docs where we described these

File: docs/resources/widgets/pages/pages.md
Comment (leighajarett) [inline docs/resources/widgets/pages/pages.md:73]: This should be a bulleted list

File: docs/resources/widgets/pages/pages.md
Comment (leighajarett) [inline docs/resources/widgets/pages/pages.md:24]: Each of these sections feel like a lot of whitespace. I think these can be bulleted lists?

File: docs/resources/widgets/pages/pages.md
Comment (leighajarett) [inline docs/resources/widgets/pages/pages.md:15]: This line doesnt seem necessary

File: docs/resources/widgets/pages/pages.md
Comment (leighajarett) [inline docs/resources/widgets/pages/pages.md:76]: I think it makes sense to show the page itself to actually show the user what an app bar, etc look like

File: docs/resources/widgets/pages/pages.md
Comment (leighajarett) [inline docs/resources/widgets/pages/pages.md:116]: I think we should modify the css to add more spacing in between bullet points

File: docs/resources/widgets/pages/pages.md
Comment (leighajarett) [inline docs/resources/widgets/pages/pages.md:106]: A visualization would probably be helpful here

File: docs/resources/widgets/pages/pages.md
Comment (leighajarett) [inline docs/resources/widgets/pages/pages.md:115]: People may not know what controllers are. Can you explain this a bit more?

File: docs/resources/widgets/pages/pages.md
Comment (leighajarett) [inline docs/resources/widgets/pages/pages.md:113]: I'm a bit confused as to what to make of this section. What do we want the user to take away from here? 

I think we probably want the user to understand when a page is reloaded (could be from the code or user interaction like refreshing a page) and what happens when its reloaded (i.e. page state variables are reset to their initial value) -- I feel like we're not providing enough helpful info here

File: docs/resources/widgets/pages/pages-configuration.md
Comment (leighajarett) [inline docs/resources/widgets/pages/pages-configuration.md:12]: Are we going to talk about configuring widgets anywhere too?

I feel like we need property panel info to be somewhere more generic, not specific to pages

File: docs/resources/widgets/pages/pages-configuration.md
Comment (leighajarett) [inline docs/resources/widgets/pages/pages-configuration.md:55]: These can be bullets

File: docs/resources/widgets/pages/how-tos.md
Comment (leighajarett) [inline docs/resources/widgets/pages/how-tos.md:7]: All pages should have an h1

File: docs/resources/widgets/pages/how-tos.md
Comment (leighajarett) [inline docs/resources/widgets/pages/how-tos.md:11]: It feels a bit strange to me that were talking about creating an empty page after we've already shown the user how to configure the page

Comment (leighajarett) [reply docs/resources/widgets/pages/how-tos.md:11]: Maybe we move the configurations to be a part of this how to guide

File: docs/resources/widgets/pages/how-tos.md
Comment (leighajarett) [inline docs/resources/widgets/pages/how-tos.md:2]: I would give it a more descriptive name like "How to use pages"

File: docs/resources/widgets/pages/how-tos.md
Comment (leighajarett) [inline docs/resources/widgets/pages/how-tos.md:163]: I think these can be bullets

File: docs/resources/widgets/pages/how-tos.md
Comment (leighajarett) [inline docs/resources/widgets/pages/how-tos.md:34]: There is some weird spacing after the arcades on this page. Seems like we need some padding or something

**Review by leighajarett — COMMENTED** [review]
Added some comments - the content itself is looking really good. Most of my comments have to do with the order in which we present the content and tweaking how the pages look so its easier to make use of the content itself

File: docs/resources/widgets/pages/how-tos.md
Comment (leighajarett) [inline docs/resources/widgets/pages/how-tos.md:353]: Feels like this can be a bulleted list

File: docs/resources/widgets/pages/how-tos.md
Comment (leighajarett) [reply docs/resources/widgets/pages/how-tos.md:2]: This page has a lot of information in it. But I worry we separated out the concepts and how-tos too much. It feels like we need to mesh them together more. I think we need to take another stab at organizing this information in a better way. 

For example, maybe having one page thats like "getting started with pages" where we describe what a page is / the scaffold thing, etc. and then show how to create a page, and configure it. 

Next page could be "navigating across pages" where we talk a bit about navigation and then the how-tos for navigation

... basically I think we give too much conceptual information and then all the how-tos here, it makes it hard to grasp the concepts and then apply them

Comment (leighajarett) [reply docs/resources/widgets/pages/how-tos.md:2]: PLus I think there's just too much on this page, we need to break it up a bit

**Review by leighajarett — COMMENTED** [review]
Left a bunch of comments. The cntent itself is great, but I dont love the order in which we display the content and how it's organized. Let me know if you wanna discuss over a call

File: docs/resources/widgets/pages/pages.md
Comment (PoojaB26) [reply docs/resources/widgets/pages/pages.md]: Yep, corrected!

File: docs/resources/widgets/pages/pages-configuration.md
Comment (PoojaB26) [reply docs/resources/widgets/pages/pages-configuration.md]: Yep, added!

**Comment by PoojaB26** [comment]
Hey @leighajarett, this is ready for review! Any missing pages or whole content, most probably is part of following PRs but you can mention it just to keep track of it!

**Review by leighajarett — CHANGES_REQUESTED** [review]
This looks amazing! 

I think we could add some how-to / arcades to the widget configuration  - should we add those in this PR or follow up?

**Review by leighajarett — CHANGES_REQUESTED** [review]
This looks amazing! 

I think we could add some how-to / arcades to the widget configuration  - should we add those in this PR or follow up?

**Comment by PoojaB26** [comment]
> This looks amazing!
> 
> I think we could add some how-to / arcades to the widget configuration - should we add those in this PR or follow up?

@leighajarett Let's do a follow up PR after this and merge this first? I can add a task to Linear. If its all good, please approve this PR then, thank you 🙌🏽

**Review by leighajarett — APPROVED** [review]

**Pull request merged**
